MovieDB/Repository/FilmRepository.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  FilmRepository.swift
+//  MovieDB
+//
+//  Created by le.n.t.trung on 09/11/2022.
+//
+
+import Foundation
+
+final class FilmRepository: RepositoryType {
+
+    typealias G = Genre
+    typealias T = DomainInfoFilm
+    typealias K = DetailInfoFilm
+
+    private let network = APICaller.shared
+
+    func getAllFilm(urlApi: String, completion: @escaping ([DomainInfoFilm]?, Error?) -> Void) {
+        network.getJSON(urlApi: urlApi, method: MethodRequest.get.rawValue) { (data: FilmList?, _) in
+            var filmList = [DomainInfoFilm]()
+            if let data = data {
+                for film in data.results {
+                    filmList.append(DomainInfoFilm(
+                        id: film.id,
+                        title: film.title,
+                        posterImageURL: film.posterPath,
+                        isAdult: film.adult,
+                        voteAverage: film.voteAverage,
+                        genre: film.genreIds,
+                        genreString: []
+                    ))
+                }
+                DispatchQueue.main.async {
```

Comment:
```suggestion
            completion(data?.result, nil)
```
Thiếu handle error đúng k e ?